barretenberg/cpp/src/barretenberg/crypto/pedersen_hash/pedersen.cpp

```diff
@@ -30,16 +30,14 @@ std::vector<typename Curve::BaseField> pedersen_hash_base<Curve>::convert_buffer
     };
 
     std::vector<Fq> elements;
-    for (size_t i = 0; i < num_elements; ++i) {
-        size_t bytes_to_slice = 0;
-        if (i == num_elements - 1) {
-            bytes_to_slice = num_bytes - (i * bytes_per_element);
-        } else {
-            bytes_to_slice = bytes_per_element;
-        }
+    for (size_t i = 0; i < num_elements - 1; ++i) {
```

Comment:
maybe double check this change

Reply:
looks fine both here and in poseidon but I don't see any unit tests for `hash_buffer` can you add some please?

Reply:
yes, good point, will add those